core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableHashJoin.java

```diff
@@ -229,7 +229,10 @@ private Result implementHashJoin(EnumerableRelImplementor implementor, Prefer pr
                         Expressions.constant(joinType.generatesNullsOnLeft()))
                     .append(
                         Expressions.constant(
-                            joinType.generatesNullsOnRight())))).toBlock());
+                            joinType.generatesNullsOnRight()))
```

Comment:
I was expecting `BuiltInMethod.java` to be modified in this patch, in order to add the new boolean parameter (isConditionAlwaysTrue) to the `HASH_JOIN` declaration. To my surprise, there was already a mismatch, because right now BuiltInMethod.HASH_JOIN is defined with parameters Enumerable, Function1, Function1, Function2:
`HASH_JOIN(ExtendedEnumerable.class, "hashJoin", Enumerable.class, Function1.class, Function1.class, Function2.class),`
but here in EnumerableHashJoin we are creating a hashJoin call expression with parameters Enumerable, Function1, Function1, Function2, **EqualityComparer, boolean, boolean** (and now yet another boolean with the current patch).
In the end it seems to work fine anyway, I guess because ExtendedEnumerable defines several overloaded hashJoin methods, but IMHO we should update BuiltInMethod.HASH_JOIN to reflect on its parameter list the actual types that are used when generating the hashJoin expression in EnumerableHashJoin, i.e.:
`HASH_JOIN(ExtendedEnumerable.class, "hashJoin", Enumerable.class, Function1.class, Function1.class, Function2.class, EqualityComparer.class, boolean.class, boolean.class, boolean.class),`

Reply:
Well, i only found one usage of `BuiltInMethod.HASH_JOIN`, so i'm a little confused why we have 3 (now 4) `ExtendedEnumerable#hashJoin` methods.

Broadly i agree to add full arguments type to `BuiltInMethod#HASH_JOIN`, thanks @rubenada !